### ISSUE TITLE
Questions on how to compute the metric given a trained model

### ISSUE THREAD

**Issue by Yuanchen-Wu** [anpwu/CauGramer#1]
Hi Anpeng,

Congrats on the paper getting accepted by ICLR! While working on replicating the results of the paper, I am a little bit bit confused with how to compute the evaluation metric for calculating IME, IPE and ITE. Let's say you have a CauGamer class trained model: 

y_0,y_1,y_o,rep,att,p_hat=model(adj,x,t,PNum)

If possible, I would like to know how to proceed with calculating IME, IPE and ITE based on your definition in the paper. It would be wonderful if you can provide some insights on how to compute them. Thank you so much!

Yuanchen

**Comment by Yuanchen-Wu** [comment]
For example, this is my current implementation of individual main effect estimate based on your paper. I wonder if this is the correct understanding of your definition? Appreciate your help!

t_0 = torch.zeros(n, device=x.device)
with torch.no_grad():
    _, _, y_ime0, _, _, _ = mymodel(adj, x, t_0, PNum)

IME_est = torch.zeros(n)
for i in range(n):
    t_1 = torch.zeros(n, device=x.device)
    t_1[i] = 1
    with torch.no_grad():
        _, _, y_ime1, _, _, _ = mymodel(adj, x, t_1, PNum)
    IME_est[i] = y_ime1[i].cpu() - y_ime0[i].cpu()

**Comment by anpwu** [comment]
Thanks for the question! Based on the model’s predictions:
```python
Tzero = T - T

pred_Y00, pred_Y10, _, _, _, _ = self.model(A, X, Tzero, PNum)
pred_Y0N, pred_Y1N, _, _, _, _ = self.model(A, X, T, PNum)

pred_YT0 = T * pred_Y10 + (1 - T) * pred_Y00
pred_YTN = T * pred_Y1N + (1 - T) * pred_Y0N

individual_effect, ATE = self.get_peheLoss(pred_YT0, pred_Y00, YT0, Y00)
peer_effect, PTE = self.get_peheLoss(pred_Y0N, pred_Y00, Y0N, Y00)
total_effect, TTE = self.get_peheLoss(pred_YTN, pred_Y00, YTN, Y00)
```

**Comment by Yuanchen-Wu** [comment]
Thanks for the clarification!
